django/middleware/csp.py

```diff
@@ -0,0 +1,124 @@
+import base64
+import os
+from functools import partial
+
+from django.conf import settings
+from django.utils.deprecation import MiddlewareMixin
+from django.utils.functional import SimpleLazyObject
+
+# TODO: Do we want to exclude CSP from a default set of views in DEBUG mode?
```

Comment:
The django-csp project disables CSP headers when DEBUG is True for 404 and 500 responses, as Django's debug views may require features like inline styles that CSP could block (I haven't actually checked on this). Since DEBUG mode isn't for production, this seems reasonable. For other response codes, applying CSP headers in DEBUG mode helps with local testing and development.